pkg/cli/cli.go

```diff
@@ -389,55 +397,82 @@ func (c cli) buildRootCmd() *cobra.Command {
 
 // defaultCommand returns the root command without its subcommands.
 func (c cli) defaultCommand() *cobra.Command {
-	return &cobra.Command{
-		Use:   c.commandName,
-		Short: "Development kit for building Kubernetes extensions and tools.",
-		Long: fmt.Sprintf(`Development kit for building Kubernetes extensions and tools.
-
-Provides libraries and tools to create new projects, APIs and controllers.
-Includes tools for packaging artifacts into an installer container.
+	cmd := &cobra.Command{}
+	cmd.Use = c.cmdCfg.CommandName
+	if cmd.Use == "" {
+		cmd.Use = defaultCommandName
+	}
+	//Get default descriptions for root commands
+	sh, lo, ex := c.getRootCommandDefaults()
 
-Typical project lifecycle:
+	cmd.Short = c.cmdCfg.Short
+	if cmd.Short == "" {
+		cmd.Short = sh
+	}
 
-- initialize a project:
+	cmd.Long = c.cmdCfg.Long
```

Comment:
```suggestion
	cmd.Long = fmt.Sprintf(c.cmdCfg.Long, cmd.Use)
```

Reply:
Setting the fields early, may not give the intended results with this. If the user-supplied string doesn't have any format specifiers specified with `%[1]s`, we will get a syntax error of the kind `(MISSING)(EXTRA string=cmd.Use)`. 
Then to resolve this case we might need to include a function to parse user strings first and then pass to variadic `Sprintf` based on results of parsing.

What do you think?

Reply:
Users are setting this value, so `c.cmdCfg.Long` should be used without modification.